cmd/otel-allocator/config/config.go

```diff
@@ -44,17 +44,19 @@ const (
 )
 
 type Config struct {
-	ListenAddr             string                `yaml:"listen_addr,omitempty"`
-	KubeConfigFilePath     string                `yaml:"kube_config_file_path,omitempty"`
-	ClusterConfig          *rest.Config          `yaml:"-"`
-	RootLogger             logr.Logger           `yaml:"-"`
-	CollectorSelector      *metav1.LabelSelector `yaml:"collector_selector,omitempty"`
-	PromConfig             *promconfig.Config    `yaml:"config"`
-	AllocationStrategy     string                `yaml:"allocation_strategy,omitempty"`
-	FilterStrategy         string                `yaml:"filter_strategy,omitempty"`
-	PrometheusCR           PrometheusCRConfig    `yaml:"prometheus_cr,omitempty"`
-	PodMonitorSelector     map[string]string     `yaml:"pod_monitor_selector,omitempty"`
-	ServiceMonitorSelector map[string]string     `yaml:"service_monitor_selector,omitempty"`
+	ListenAddr                      string                `yaml:"listen_addr,omitempty"`
+	KubeConfigFilePath              string                `yaml:"kube_config_file_path,omitempty"`
+	ClusterConfig                   *rest.Config          `yaml:"-"`
+	RootLogger                      logr.Logger           `yaml:"-"`
+	CollectorSelector               *metav1.LabelSelector `yaml:"collector_selector,omitempty"`
+	PromConfig                      *promconfig.Config    `yaml:"config"`
+	AllocationStrategy              string                `yaml:"allocation_strategy,omitempty"`
+	FilterStrategy                  string                `yaml:"filter_strategy,omitempty"`
+	PrometheusCR                    PrometheusCRConfig    `yaml:"prometheus_cr,omitempty"`
+	PodMonitorSelector              map[string]string     `yaml:"pod_monitor_selector,omitempty"`
+	ServiceMonitorSelector          map[string]string     `yaml:"service_monitor_selector,omitempty"`
+	ServiceMonitorNamespaceSelector map[string]string     `yaml:"service_monitor_namespace_selector,omitempty"`
+	PodMonitorNamespaceSelector     map[string]string     `yaml:"pod_monitor_namespace_selector,omitempty"`
```

Comment:
Could we make the new fields into a `metav1.LabelSelector`, just like `CollectorSelector`? We want every selector to use that format eventually.

Reply:
@swiatekm-sumo - Updated. Please take a look.